### ISSUE TITLE
vtkPlane's intersectWithLine() function return is mathematically confusing

### ISSUE THREAD

**Issue by mix3d** [Kitware/vtk-js#1189]
the function is given a line defined by two points, but does not specify that the intersection HAS to happen between those two points for the result to return `intersect:true`.

A line defined by two points extends BEYOND those two points, and can still intersect, but if the intersection is outside of the defined points, you get a return with `intersect:false` but still get the coordinates of the intersection.

I think the function & corresponding return is misleading.

**Comment by mix3d** [comment]
This will conflict with the new `intersectWithPlane()` function that returns a line defined by two points, which COULD intersect with a 3rd plane, but if you assume the `result.intersect` will only be false if the result is parallel, you will get false negatives in your logic.

**Comment by mix3d** [comment]
TLDR: the `result.intersect` value is used both to say "Line and Plane are parallel (false)" and "Plane intersects line between the line's provided points (true / false)"

**Comment by mix3d** [comment]
I have no problem committing a fix, but want to be clear that it would be a breaking change.

**Comment by finetjul** [comment]
I agree, it is a bit misleading. 
However, it is the same behavior than VTK (C++).

**Comment by finetjul** [comment]
Maybe `intersectWithLine(p0,p1)` could be renamed `intersectWithSegment(p0, p1)` and documented as behaving like `VTK(C++) intersectWithLine` and add new `intersectWithLine` as such:
```
function intersectWithLine(p0, p1) {
  res = vtkPlane.intersectWithSegment(p0, p1);
  return Object.assign(res, {intersect: res.intersect || res.t < Number.MAX_VALUE });
```
